client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -247,7 +258,7 @@ class SimplePaymentsDialog extends Component {
 		this.setIsSubmitting( true );
 
 		const { siteId, dispatch, translate } = this.props;
-
+		analytics.tracks.recordEvent( 'calypso_simple_payments_button_delete', { id: paymentId } );
```

Comment:
If possible, could we use analytics middleware here? We try to avoid direct analytics lib calls for testing purposes. In this case it'd be:
```jsx
import { withAnalytics, recordTracksEvent } from 'state/analytics/actions';

//...

dispatch( withAnalytics(
    recordTracksEvent( 'calypso_simple_payments_button_delete' ),
    trashPaymentButton( //... )
) );
```